src/main/jbake/content/blog/2015/04/14/hawkular-monitor-agent-intro.adoc

```diff
@@ -0,0 +1,106 @@
+
+A sample configuration https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-monitor/src/main/assembly/subsystem.xml[is found here].
+
+An example of a metric to be collected is:
+
+    <metric-dmr name="heap.used"
+                interval="30"
+                timeUnits="seconds"
+                resource="/core-service=platform-mbean/type=memory"
+                attribute="heap-memory-usage#used" />
+
+This defines a metric named "heap.used" which is the heap memory used by the WildFly VM.
+The metric will be collected every 30 seconds.
+
+Here is an example of an availability check:
+
+    <avail-dmr name="app-server"
+               interval="30"
+               timeUnits="seconds"
+               resource="/"
+               attribute="server-state"
+               upRegex="run.*" />
+
+This will look at the WildFly's "server-state" and if its value matches the
+regular expression "run.*" then this availability check will report UP. Otherwise, it will report DOWN.
+If no upRegex is provided, then some simple heuristics are performed to determine the availability status
+(specifically if the value is a number then a 0 is considered DOWN and non-zero is considered UP;
+otherwise, if the value is the string "up" or "ok" (case insenstive) it is considered UP,
+otherwise it is considered DOWN.)
+
+The full XML Schema for the Hawkular Monitor agent subsystem configuration can be
+https://github.com/hawkular/hawkular-agent/blob/master/hawkular-wildfly-monitor/src/main/resources/schema/hawkular-agent-monitor-subsystem.xsd[found here].
+
+Currently, what can be monitored or checked for availability is any resource in the WildFly management hierachy
+(in the same WildFly where Hawkular Monitor agent is deployed). Future development is underway to allow the
+Hawkular Monitor agent to monitor resources in remote WildFly instances, as well as remote JMX resources.
+
+Today, you have the option to store the metric and availability data directly to a Hawkular-Metrics instance or to
+a Hawkular ecosystem (this is the <storage-adapter> definition). As the Hawkular ecosystem is built out (specifically, as inventory is brought into the picture),
+a fuller integration with Hawkular Monitor agent will be possible.
+
+If you have any questions on this, or suggestions on what features we need to add or enhance, please ping us on
+the #hawkular freenode chat room or send an email to the https://lists.jboss.org/mailman/listinfo/hawkular-dev[hawkular-dev mailing list].
+
+== How To Build
+. Clone https://github.com/hawkular/hawkular-agent[the source code]
+. Build via Maven: "mvn clean install"
+* The Wildfly module that houses the Hawkular Monitor agent will be found in a zip file under hawkular-agent/hawkular-wildfly-monitor.
+. Unzip the Hawkular Monitor agent into whatever Wildfly 8.2 instance you want.
+* Alternatively, you can use the Maven wildfly-extension-plugin to do this:
+** cd hawkular-agent/hawkular-wildfly-monitor
```

Comment:
If you use source code block formatting instead of list, it's easier for the reader to copy/paste the commands in a terminal:
```
[source,bash]
----
----
\\\